### PR TITLE
docs(tutorials): change ReST inline role to MyST

### PR DESCRIPTION
**Pull request by clijo**
# Issue
- Some tutorial notebooks used ReST inline role syntax for glossary terms `:term:` instead of MyST `{term}`, breaking the hyperlinks, see for instance [Point Cloud](https://ott-jax.readthedocs.io/tutorials/geometry/000_point_cloud.html).
- Missing asterisk for tuple unpacking causes an error for $W_{\text{cosine}}$ Gradient Flow in the  [Point Cloud tutorial](https://ott-jax.readthedocs.io/tutorials/geometry/000_point_cloud.html#ot-gradient-flows).

# Fix
- Instances of `:term:` in notebooks replaced with `{term}`; now correctly link to the terms in glossary.
- Fixed the tuple unpacking error for Point Cloud tutorial. This required rerunning the notebook, hence the large diff.

### CODE REVIEW

**Comment by michalk8** [comment]
LGTM, thanks a lot!
